secure/storage/vault/src/lib.rs

```diff
@@ -26,8 +27,9 @@ pub mod fuzzing;
 /// Keys are trimmed in FIFO order.
 const MAX_NUM_KEY_VERSIONS: u32 = 4;
 
-/// Request timeout for vault operations
-const TIMEOUT: u64 = 10_000;
+/// Request timeouts for vault operations.
+const CONNECT_TIMEOUT_MILLISECS: u64 = 3000;
+const TIMEOUT_MILLISECS: u64 = 5000;
```

Comment:
@mgorven, @davidiw, you may have to come to some sort of compromise regarding these values 😄

Reply:
How much work would it be to make these configurable?

Reply:
Per component? e.g., using the config files, I believe a PR was previously started last year but never landed (https://github.com/diem/diem/pull/5229). Given the amount of code for this, do we think configurable timeouts would be that useful?

Reply:
In the Vault `backend` config section (wherever that gets used). Yes, timeouts are almost always configurable and do get used :-) If they are configurable we can set fairly aggressive defaults, but if that causes issues we can tune them in deployments. If they're hardcoded they have to be much more conservative.

Reply:
That's fair -- if you can see the benefit then I'm happy to do that. Will probably have to do it as a follow up though -- might be too far beyond the scope of this PR 😄

Reply:
I think you can attach it as another commit to this PR or you can keep the values at 10_000.
I'm not comfortable changing values without having knobs at this point.

Reply:
Agreed, I've changed them to 10_000 and will then follow up with a configurable timeout PR (I think this will be cleaner).